packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3442,6 +3459,14 @@ class InputDecorationTheme extends Diagnosticable {
   /// Defaults to true.
   final bool hasFloatingPlaceholder;
 
+  /// Whether the label is always floated.
+  ///
+  /// If this is false, the placeholder label floats when the input has focus or text has been entered.
```

Comment:
I think this should be more clear about what floating is.  Maybe like:

"When true, the label will always be at the top of the field above the content.  When false, the label will float to the top only when the field is focused or has some text content, otherwise it will appear in the field in place of the content."